hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/RestoreSnapshotProcedure.java

```diff
@@ -70,6 +71,7 @@ public class RestoreSnapshotProcedure
   extends AbstractStateMachineTableProcedure<RestoreSnapshotState> {
   private static final Logger LOG = LoggerFactory.getLogger(RestoreSnapshotProcedure.class);
 
+  private TableDescriptor oldTableDescriptor;
```

Comment:
This variable needs to be added to `RestoreSnapshotStateData` defined in `hbase-protocol-shaded/src/main/protobuf/server/master/MasterProcedure.proto`

Also the variable needs to be added in `serializeStateData()` and `deserializeStateData()` of this class. 

It will throw NPE in case master failover happens before it reaches `RESTORE_SNAPSHOT_UPDATE_TABLE_DESCRIPTOR` or `RESTORE_SNAPSHOT_SYNC_ERASURE_CODING_POLICY` state as `oldTableDescriptor` will become null with serialize/deserialization .

Reply:
Thanks for catching this. Fixed